rs/index/src/collection/collection.rs

```diff
@@ -346,6 +347,20 @@ impl<Q: Quantizer + Clone + Send + Sync + 'static> Collection<Q> {
                                 .unwrap();
                         });
                 }
+                WalOpType::Delete => {
+                    info!("Processing delete operation with seq_no {}", op.seq_no);
+                    let doc_ids = op.doc_ids();
+                    let user_ids = op.user_ids();
+                    assert!(op.data().is_empty());
+                    user_ids.iter().try_for_each(|&user_id| {
+                        doc_ids.iter().try_for_each(|&doc_id| {
+                            self.remove(user_id, doc_id)?;
+                            Ok(())
+                        })
+                    })?;
+                }
+                // For future operations
```

Comment:
let's actually remove the default branch, so that when we add a new op, compiler will fail to compile